File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.java
class Solution {
    public int[] solution(long begin, long end) {
        long size = end - begin + 1;
        int[] answer = new int[(int) size];

        for (long i = begin; i <= end; i++) {
            if (i == 1)
            {
                answer[0] = 0;
                continue;
            }
            for (int j = (int)Math.sqrt(i); j > 0; --j) {
                
                if (i % j == 0) {
                    if (j == 1)
                    {
                        answer[(int)(i - begin)] = Math.max(answer[(int)(i - begin)], 1);
                        break;
                    }
                    
                    if (i / j > 10000000)
                    {
                        answer[(int)(i - begin)] = Math.max(answer[(int)(i - begin)], j);
                        continue;
                    }
                    
                    
                    answer[(int)(i - begin)] = (int)i / j;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int m = 10000000;

vector<int> solution(long long begin, long long end) {
    vector<int> answer (end - begin + 1, 0);

    for (int i = begin; i <= end; ++i)
    {
        if (i == 1)
        {
            answer[i - begin] = 0;
            continue;
        }

        for (int j = sqrt(i); j > 0; --j)
        {
            if (i % j == 0)
            {
                if (j == 1)
                {
                    answer[i - begin] = max(answer[i-begin], 1);
                    break;
                }
                if (i / j > m)
                {
                    answer[i - begin] = max(answer[i - begin], j);
                    continue;
                }

                answer[i - begin] = i / j;
            }
        }

    }

    return answer;
}